application/src/main/java/org/thingsboard/server/service/edge/DefaultEdgeNotificationService.java

```diff
@@ -157,79 +157,78 @@ public Edge setEdgeRootRuleChain(TenantId tenantId, Edge edge, RuleChainId ruleC
+                            break;
+                        case ENTITY_VIEW:
+                            entityViewProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case DASHBOARD:
+                            dashboardProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case RULE_CHAIN:
+                            ruleChainProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case USER:
+                            userProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case CUSTOMER:
+                            customerProcessor.processCustomerNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case DEVICE_PROFILE:
+                            deviceProfileProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case ASSET_PROFILE:
+                            assetProfileProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case OTA_PACKAGE:
+                            otaPackageProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case WIDGETS_BUNDLE:
+                            widgetBundleProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case WIDGET_TYPE:
+                            widgetTypeProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case QUEUE:
+                            queueProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case ALARM:
+                            alarmProcessor.processAlarmNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case RELATION:
+                            relationProcessor.processRelationNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case TENANT:
+                            tenantEdgeProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        case TENANT_PROFILE:
+                            tenantProfileEdgeProcessor.processEntityNotification(tenantId, edgeNotificationMsg);
+                            break;
+                        default:
+                            log.warn("[{}] Edge event type [{}] is not designed to be pushed to edge", tenantId, type);
+                    }
```

Comment:
With thread pool size 4, you can afford tell success inside submitted task and maintain the same processing time out as CoreConsumer.